egs/babel/s5d/local/run_asr_segmentation.sh

```diff
@@ -0,0 +1,336 @@
+# Get targets from lattices
+###############################################################################
+if [ $stage -le 6 ]; then
+  steps/segmentation/lats_to_targets.sh --cmd "$train_cmd" \
+    --silence-words $dir/silence_words.txt \
+    --garbage-words $dir/garbage_words.txt \
+    --max-phone-duration 0.5 \
+    $data_dir $lang $sup_lats_dir \
+    $dir/${model_id}_${data_id}_sup_targets
+fi
+
+if [ $stage -le 7 ]; then
+  steps/segmentation/lats_to_targets.sh --cmd "$train_cmd" \
+    --silence-words $dir/silence_words.txt \
+    --garbage-words $dir/garbage_words.txt \
+    --max-phone-duration 0.5 \
+    $uniform_seg_data_dir $lang $decode_dir \
+    $dir/${model_id}_${uniform_seg_data_id}_targets
+fi
+
+###############################################################################
+# Convert targets to be w.r.t. whole data directory
+###############################################################################
+if [ $stage -le 8 ]; then
+  echo " [ 1 0 0 ]" > $dir/${model_id}_${whole_data_id}_sup_targets/default_targets.vec
+  
+  steps/segmentation/convert_targets_dir_to_whole.sh --cmd "$train_cmd" --nj 40 \
+    --default-targets $dir/${model_id}_${whole_data_id}_sup_targets/default_targets.vec \
+    $data_dir $whole_data_dir \
+    $dir/${model_id}_${data_id}_sup_targets \
+    $dir/${model_id}_${whole_data_id}_sup_targets
+  
+  steps/segmentation/resample_targets_dir.sh --cmd "$train_cmd" --nj 40 3 \
+    $whole_data_dir \
+    $dir/${model_id}_${whole_data_id}_sup_targets \
+    $dir/${model_id}_${whole_data_id}_sup_targets_sub3
+fi
+
+###############################################################################
+# Resample targets to required output-subsampling rate
+###############################################################################
+if [ $stage -le 9 ]; then
+  steps/segmentation/convert_targets_dir_to_whole.sh --cmd "$train_cmd" --nj 40 \
+    $dir/${uniform_seg_data_id} $whole_data_dir \
+    $dir/${model_id}_${uniform_seg_data_id}_targets \
+    $dir/${model_id}_${whole_data_id}_targets
+
+  steps/segmentation/resample_targets_dir.sh --cmd "$train_cmd" --nj 40 3 \
+    $whole_data_dir \
+    $dir/${model_id}_${whole_data_id}_targets \
+    $dir/${model_id}_${whole_data_id}_targets_sub3
+fi
+
+merge_weights=1.0,0.1
+###############################################################################
+# Merge targets for the same data from multiple sources (systems)
+###############################################################################
+if [ $stage -le 10 ]; then
+  steps/segmentation/merge_targets_dirs.py --cmd "$train_cmd" --nj 40 \
+    --weights $merge_weights --remove-mismatch-frames=true \
+    $whole_data_dir \
+    $dir/${model_id}_${whole_data_id}_sup_targets_sub3 \
+    $dir/${model_id}_${whole_data_id}_targets_sub3 \
+    $dir/${model_id}_${whole_data_id}_combined_targets_sub3
+fi
+
+
+mkdir -p data/lang_asr_sad_fs3_simple
```

Comment:
I think a paragraph of comment at this point would be helpful, explaining what these mean.